CMakeLists.txt

```diff
@@ -134,6 +134,20 @@ if(${OQS_USE_OPENSSL})
         endif()
     endif()
     find_package(OpenSSL 1.1.1 REQUIRED)
+
+    if(OQS_DLOPEN_OPENSSL)
+      find_program(OBJDUMP objdump)
```

Comment:
This seems to imply that this can only work on UNIX'ish systems, right (doc mention of this fact if I got right as per the previous comment, please)? I also infer this from absent Windows tests. Thanks in turn for adding Unix CI for this new feature.

Reply:
I think this should work on ELF-based Unix platforms (Linux, BSD, etc.), where objdump whose output is compatible with GNU binutils is installed.

Reply:
OK -- could you then please state it suitably in the config variable documentation for folks not checking out the code? I'm just wondering about MacOS (?).

Reply:
Elaborated that in the documentation.

macOS is a bit tricky as it's [Mach-O](https://en.wikipedia.org/wiki/Mach-O) based, where I'm not sure if SONAME is as part of the ABI, and uses different tooling (e.g., otool instead of ldd/objdump).